pkg/storage/stores/shipper/indexshipper/downloads/table.go

```diff
@@ -148,6 +162,10 @@ func (t *table) Close() {
 
 func (t *table) ForEachConcurrent(ctx context.Context, userID string, callback index.ForEachIndexCallback) error {
 	g, ctx := errgroup.WithContext(ctx)
+	if t.maxConcurrent == 0 {
+		panic("maxConcurrent cannot be 0, downloads.table is being initialized without setting maxConcurrent")
+	}
+	g.SetLimit(t.maxConcurrent)
```

Comment:
nit: should we skip setting the limit here? we almost always spawn 2 routines unless it's a multi-tenant query